### ISSUE TITLE
CJavaScript::quote(null); throws error, incompatible change since Yii 1.1.20

### ISSUE THREAD

**Issue by stevoh6** [yiisoft/yii#4238]
Note that only PHP 7 compatibility issues are accepted. For security issues contact maintainers privately.

### Background information
Since many applications running on Yii1 are quite old, we are facing one backward incompatible change. 

I'm suggesting add simple check, pull request created. 

### What steps will reproduce the problem?
Using 
`CJavaScript::quote(null); `
throws error.
### What is the expected result?
Do not try escaping/converting null value as it was before https://github.com/yiisoft/yii/commit/1e1c5b7b058eedf3eb2f58914b6be35e965463fc#diff-34b1d76ab6620e419eaef5799cff7103. 
### What do you get instead?
`Exception
String to be escaped was not valid UTF-8 or could not be converted:
/var/www/app/protected/vendor/yiisoft/yii/framework/vendors/zend-escaper/Escaper.php(321)`

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 1.1.20
| PHP version      | 7.2
| Operating system | Alpine

**Comment by rob006** [comment]
> Using
> `CJavaScript::quote(null); `
> throws error.

This looks like a bug in your app. Why would you want to quote `null`?

**Comment by samdark** [comment]
It was never type-hinted to accept null so I'd like to hear the reason as well.

**Comment by stevoh6** [comment]
Hi, I fully understand, that this is unusual usage this function, I was take over couple old projects and it was pretty surprise for me how it was implemented in our projects, but I get the idea why. Of course it can be rewrite and Im already make some adjustment to avoid this "behavior", but it would be to nice don't change old behavior. So I decided open pull request, maybe someone have similar "problem" :-)

One of real (not my, not really good) example: from controller you get some settings values (stored in db), you pass this whole array to view and you trying put those values into js variables (for example to fill input-boxes) and you would like to quoted them regardless their value.
